pkg/apis/workflow/v1alpha1/artifact_repository_types.go

```diff
@@ -133,6 +137,22 @@ func (r *ArtifactoryArtifactRepository) IntoArtifactLocation(l *ArtifactLocation
 	l.Artifactory = &ArtifactoryArtifact{ArtifactoryAuth: r.ArtifactoryAuth, URL: u}
 }
 
+// AzureBlobArtifactRepository defines the controller configuration for an Azure Blob Storage artifact repository
+type AzureBlobArtifactRepository struct {
```

Comment:
Would AzureArtifactRepository be simpler or is there any concern in ambiguity?

Reply:
@terrytangyuan this change has been made. Please let me know if there's anything else. Thanks!